compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -457,7 +487,16 @@ impl<'a, 'tcx> AbstractConstBuilder<'a, 'tcx> {
             if let Some(next) = self.build_terminator(block.terminator())? {
                 block = &self.body.basic_blocks()[next];
             } else {
-                return Ok(self.tcx.arena.alloc_from_iter(self.nodes));
+                assert_eq!(self.locals[mir::Local::from_usize(0)], self.nodes.last().unwrap());
+                self.nodes[self.locals[mir::Local::from_usize(0)]].1 = true;
+                if !self.nodes.iter().all(|n| n.1) {
+                    self.error(None, "unused node")?;
+                }
+
+                return Ok(self
```

Comment:
if you use a separate map for metadata not needed for the result, you can use `alloc_slice` which is cheaper than `alloc_from_iter` (though that will probably need a method to get a slice from an `IndexVec`, too)

Reply:
I don't think perf is too important here and prefer to prevent accidents where we forget to update one of the vecs, will add spans though